### PR TITLE
CW: snap mode preparations #1

### PR DESCRIPTION
**Pull request by DD4WH**
CW Menu:
- enable CW decoder
- enable "carrier snap"
- change Demod_mode to CW
- adjust threshold, so that CW station modulates the red LED
- the small frequency display shows the determined carrier frequency in green
- be a little patient until frequency display has settled
TODO: graphical display to show offset and to help in zero beat tuning

### CODE REVIEW

File: mchf-eclipse/drivers/audio/audio_driver.h [hunk 1]
```diff
@@ -193,6 +193,9 @@ typedef struct AudioDriverState
     float32_t               iq_phase_balance_rx;
     float32_t               iq_phase_balance_tx[IQ_TRANS_NUM];
 
+    ulong snap_carrier_freq; // used for passing the estimated carrier freq in SNAP mode to the print routine in UI_Driver
+    bool CW_signal;
```
Comment (db4ple) [inline mchf-eclipse/drivers/audio/audio_driver.h:197]: Please document the purpose of this variable. Thank you!

File: mchf-eclipse/drivers/ui/ui_driver.c [hunk 2]
```diff
@@ -6370,6 +6371,10 @@ void UiDriver_MainHandler()
 				{
 					UiDriver_UpdateLcdFreq(df.tune_old/TUNE_MULT, Yellow, UFM_SECONDARY);
 				}
+				else if (ts.dmod_mode == DEMOD_CW)
+				{
+					UiDriver_UpdateLcdFreq(ads.snap_carrier_freq, Green, UFM_SECONDARY);
+				}
```
Comment (db4ple) [inline mchf-eclipse/drivers/ui/ui_driver.c:6377]: I think you are not going to make friends with the RIT users here, I believe RIT is being used in CW.  
My comment re automatic RIT is not right, done one function above. So you just kill the RIT display as stated above.

Comment (DD4WH) [reply mchf-eclipse/drivers/ui/ui_driver.c:6377]: Well, I do not see a problem here: RIT users can switch off this option ;-). Additionally, the currrent green frequency display is just to show that carrier frequency estimation is possible in this first alpha version. The display will be no longer useful once the graphical offset indicator is ready.

**Comment by df8oe** [comment]
RIT is intensively used in CW.

**Comment by DD4WH** [comment]
Well, as I said: RIT users can switch this option OFF and please also note (as stated above) that this is temporary display until the graphical display is ready. So please be patient ;-).

**Comment by df8oe** [comment]
Frequency display does not change to RIT if leaving snap mode...

**Pull request merged**